### ISSUE TITLE
podman wait fails with 'error joining network namespace'

### ISSUE THREAD

**Issue by edsantiago** [projectatomic/libpod#737]
This is another irreproducible one: I've only seen it twice in the last month. The common recipe is:
```
# podman start simple_l7j8     ! Expected to fail. I can't imagine this step being necessary but it's part of the test.
# podman run --interactive=true --name simple_l7j8 registry.access.redhat.com/rhel7/rhel:latest bash -c 'echo STARTED: $(date); while :; do sleep 0.1; done'
```
In another window:
```
# podman start simple_l7j8   ! NOP. Again, part of the test, dunno if it's part of the failure
# podman kill simple_l7j8
# podman wait simple_l7j8
unable to find container simple_l7j8: error joining network namespace for container cb267e5f18b7a39354bf9ab717c9f1694ae0b1ba1774f4e51be084f1e6180b95: error retrieving network namespace at /var/run/netns/cni-0fd25158-8374-8a3f-7939-a62c624b0426: unknown FS magic on "/var/run/netns/cni-0fd25158-8374-8a3f-7939-a62c624b0426": 1021994
```
The SHAs and UUIDs differ between the two failures but the `1021994` is the same.

podman-0.5.2-1525700809.gitfaa8c3e.fc27.x86_64

**Comment by mheon** [comment]
So it's `kill` then `wait` that does it...

This is potentially a race around network namespace removal. `podman kill` stops the container, causing `podman run`'s cleanup routines to fire. `podman wait` tries to read the container out of the database while the cleanup routines are in the process of deleting the network namespace but before they have committed the fact that it no longer exists to the DB. `wait` gets a bad namespace (it's half-deleted) and complains that it can't retrieve the container from the database because it has an invalid network namespace.

We're probably better off reversing the order of operations for network namespaces to removing them from the DB, and then deleting them from disk, as opposed to the other way around.

**Comment by rhatdan** [comment]
Opened a PR to fix this. PTAL

**Comment by mheon** [comment]
Should be fixed now that #895 is merged
